airbyte-cdk/java/airbyte-cdk/dependencies/src/main/java/io/airbyte/commons/functional/Either.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.commons.functional;
+
+import java.util.Objects;
+
+/**
+ * A class that represents a value of one of two possible types (a disjoint union). An instance of
+ * Either is an instance of Left or Right.
+ *
+ * A common use of Either is for error handling in functional programming. By convention, Left is
+ * failure and Right is success.
+ *
+ * @param <Error> the type of the left value
+ * @param <Result> the type of the right value
+ */
+public class Either<Error, Result> {
```

Comment:
we already have io.airbyte.commons.functional.Either (... git blame says you wrote it :joy:)